include/multipass/mount_config.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2019-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_MOUNT_CONFIG_H
+#define MULTIPASS_MOUNT_CONFIG_H
+
+#include <multipass/id_mappings.h>
+#include <multipass/virtual_machine.h>
+
+#include <unordered_map>
+
+namespace multipass
+{
+
+struct MountConfig
+{
+    VirtualMachine* vm;
+    id_mappings uid_mappings;
+};
```

Comment:
Thanks, this makes it much easier to understand (where this is used). 

You have it in a header with a view to use it for other mounts, correct? Do you know if this pattern fits the code for other mounts (existing or upcoming)?

Reply:
Yes, the idea was that if we have another kind of mount where we need some configuration data, we could just add additional fields to the struct.

I say that, but I don't have any insight fits the requirements for other mounts. I had decided not to merge this with [SSHFSServerConfig](https://github.com/canonical/multipass/blob/765d09de8043f69a4beaf4a9404a68cddd60cf80/include/multipass/sshfs_server_config.h#L29) as I feel SSHFS is too different with it's need to spawn a separate process for each mount.